### PR TITLE
Use Navigation Treeview pattern for RoomList Accessibility

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/11113

Video of it in action:
https://riot.ovh/_matrix/media/r0/download/riot.ovh/dsMfEvkkityMHmUaRlvqeIfG

### CODE REVIEW

File: src/components/structures/RoomSubList.js [hunk 1]
```diff
@@ -126,6 +127,74 @@ const RoomSubList = createReactClass({
...
+                ev.stopPropagation();
+                break;
+            case Key.ARROW_RIGHT: {
+                ev.stopPropagation();
+                if (this.state.hidden && !this.props.forceExpand) {
+                    // sublist is collapsed, expand it
+                    this.onClick();
+                } else if (!this.props.forceExpand) {
+                    // sublist is expanded, go to first room
+                    let element = document.activeElement;
```
Comment (bwindels) [inline src/components/structures/RoomSubList.js]: I wonder if you could avoid the !decending code path below and usage of `document.activeElement` if you did `ev.target.closest(".mx_RoomSubList_labelContainer")` here?

File: src/components/structures/RoomSubList.js [hunk 2]
```diff
@@ -255,10 +340,17 @@ const RoomSubList = createReactClass({
...
-                <AccessibleButton onClick={this.onClick} className="mx_RoomSubList_label" tabIndex={tabindex} aria-expanded={!isCollapsed}>
+            <div className="mx_RoomSubList_labelContainer" title={title} ref="header" onKeyDown={this.onHeaderKeyDown}>
+                <AccessibleButton
+                    onClick={this.onClick}
+                    className="mx_RoomSubList_label"
+                    tabIndex={0}
+                    aria-expanded={!isCollapsed}
+                    inputRef={this._headerButton}
+                    // cancel out role so this button behaves as the toggle-header of this group
+                    role="none"
```
Comment (bwindels) [inline src/components/structures/RoomSubList.js]: Just checking, so not `treeitem` as Marco suggested?

Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: Ah yes, that makes more sense :D

File: src/components/structures/RoomSubList.js (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: That certainly would clean this up, will test

File: src/components/structures/RoomSubList.js (hunk 2 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: This makes it a bit noisier but is more semantically correct so will commit it

File: src/components/structures/RoomSubList.js (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: ended up simplifying even more

File: src/components/structures/RoomSubList.js (hunk 2 above)
Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: so with this not being role=none it gets counted as an "item" inside that section so if you have 1 room in a section it'll say x out of 2. But is semantically correct 🤷‍♂

Comment (bwindels) [reply src/components/structures/RoomSubList.js]: Hmm, would be interesting to know whether browsers and platforms consistently behave like this, and from Marco if this is the behaviour he expected. I agree it's good to go with this now and get feedback on it.

**Pull request merged**